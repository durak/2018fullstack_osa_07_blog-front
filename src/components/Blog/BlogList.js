import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Card } from 'semantic-ui-react'

import { blogLike } from '../../reducers/blogReducer'
import { notify } from '../../reducers/notificationReducer'

import BlogListItem from './blogList/BlogListItem'

export const BlogList = ({ blogs, blogLike, notify }) => {

  return (
    <div>
      <h1>Blogs</h1>
      <Card.Group centered itemsPerRow="4" stackable>
        {blogs.map(blog =>
          <BlogListItem
            key={blog.id}
            blog={blog}
            blogLike={blogLike}
            notify={notify}
          />
        )}
      </Card.Group>
    </div>
  )

}

const mapStateToProps = (state) => {
  let sortedBlogs = []
    .concat(state.blogs)
    .sort((a, b) => {
      return b.likes - a.likes
    })

  return {
    blogs: sortedBlogs
  }
}

export default connect(
  mapStateToProps,
  { blogLike, notify }
)(BlogList)

BlogListItem.propTypes = {
  blogs: PropTypes.arrayOf(PropTypes.object),
  blogLike: PropTypes.func.isRequired,
  notify: PropTypes.func.isRequired
}