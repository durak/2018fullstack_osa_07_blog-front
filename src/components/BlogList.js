import React from 'react'
import { connect } from 'react-redux'

import Blog from './Blog'

const BlogList = ({ blogs, user }) => (

  <div>
    {blogs.map(blog =>
      <Blog
        key={blog.id}
        blog={blog}
        user={user}
      />
    )}
  </div>
)


const mapStateToProps = (state) => {
  let sortedBlogs = []
    .concat(state.blogs)
    .sort((a, b) => {
      return b.likes - a.likes
    })

  return {
    blogs: sortedBlogs
  }
}


export default connect(
  mapStateToProps
)(BlogList)