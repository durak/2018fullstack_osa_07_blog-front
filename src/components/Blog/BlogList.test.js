import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { BlogList } from './BlogList'
import { BrowserRouter as Router } from 'react-router-dom'

describe('<BlogList />', () => {

  const blog1 = {
    id: '1',
    title: 'test blog 1',
    author: 'test author 1',
    url: 'http://testurl.com 1',
    likes: 99,
    user: { name: 'test user full name 1', username: 'testusername 1' },
    comments: [
      { _id: '1', comment: 'some comment content' },
      { _id: '2', comment: 'some other content' }
    ]
  }

  const blog2 = {
    id: '2',
    title: 'test blog 2',
    author: 'test author 2',
    url: 'http://testurl.com 2',
    likes: 99,
    user: { name: 'test user full name 2', username: 'testusername 2' },
    comments: [
      { _id: '1', comment: 'some comment content' },
      { _id: '2', comment: 'some other content' }
    ]
  }

  const mockHandleLike = jest.fn()
  const mockNotify = jest.fn()

  it('renders correctly', () => {
    const tree = renderer.create(
      <Router>
        <BlogList
          blogs={[blog1, blog2]}
          notify={mockNotify}
          blogLike={mockHandleLike}
        />
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot()
  })
})