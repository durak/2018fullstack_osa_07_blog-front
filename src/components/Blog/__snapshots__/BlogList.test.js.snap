// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogList /> renders correctly 1`] = `
<div>
  <h1>
    Blogs
  </h1>
  <div
    className="ui centered stackable four cards"
  >
    <div
      className="ui card"
      href={undefined}
      onClick={[Function]}
    >
      <div
        className="content blurring dimmable"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="header"
        >
          <span
            className="blogtitle"
          >
            test blog 1
          </span>
        </div>
        <div
          className="meta"
        >
          <span>
            by 
            test author 1
          </span>
        </div>
        <div
          className="description"
        >
          test blog 1
        </div>
        <div
          className="ui inverted dimmer"
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <h2
              className="ui header"
            >
              <a
                href="/blogs/1"
                onClick={[Function]}
              >
                 View 
              </a>
            </h2>
          </div>
        </div>
      </div>
      <div
        className="extra content"
      >
        <span>
          <i
            aria-hidden="true"
            className="like icon blogVote"
            onClick={[Function]}
          />
          99
           votes
        </span>
        <span
          className="right floated"
        >
          <i
            aria-hidden="true"
            className="comments icon"
          />
          2
           comments
        </span>
      </div>
    </div>
    <div
      className="ui card"
      href={undefined}
      onClick={[Function]}
    >
      <div
        className="content blurring dimmable"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="header"
        >
          <span
            className="blogtitle"
          >
            test blog 2
          </span>
        </div>
        <div
          className="meta"
        >
          <span>
            by 
            test author 2
          </span>
        </div>
        <div
          className="description"
        >
          test blog 2
        </div>
        <div
          className="ui inverted dimmer"
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <h2
              className="ui header"
            >
              <a
                href="/blogs/2"
                onClick={[Function]}
              >
                 View 
              </a>
            </h2>
          </div>
        </div>
      </div>
      <div
        className="extra content"
      >
        <span>
          <i
            aria-hidden="true"
            className="like icon blogVote"
            onClick={[Function]}
          />
          99
           votes
        </span>
        <span
          className="right floated"
        >
          <i
            aria-hidden="true"
            className="comments icon"
          />
          2
           comments
        </span>
      </div>
    </div>
  </div>
</div>
`;
